dracut-saltboot/saltboot/saltboot.sh

```diff
@@ -90,6 +90,21 @@ EOT
 
 echo $MACHINE_ID > /etc/machine-id
 
+curl -s http://salt/saltboot/defaults > /tmp/defaults
+if [ \! -s /tmp/defaults ] ; then
+    busybox tftp -g -l /tmp/defaults -r defaults $BOOTSERVERADDR
+fi
+
+if [ -s /tmp/defaults ] ; then
+    [ -z "$MINION_ID_PREFIX" ] && eval `grep ^MINION_ID_PREFIX= /tmp/defaults`
+    [ -z "$DISABLE_ID_PREFIX" ] && eval `grep ^DISABLE_ID_PREFIX= /tmp/defaults`
+    [ -z "$DISABLE_UNIQUE_SUFFIX" ] && eval `grep ^DISABLE_UNIQUE_SUFFIX= /tmp/defaults`
+    if [ -z "$USE_FQDN_MINION_ID" -a -z "$DISABLE_HOSTNAME_ID" ] ; then
```

Comment:
Why is here and condition? It is true that these two together enabled doesn't make sense, but the logic determining it is in different place.

Reply:
The idea is not use the value from defaults if at least one of the variables is specified on kernel commandline. I think it could lead to unexpected combinations otherwise.

Reply:
Ah, right. I didn't think about that.